config/partitions_custom.csv

```diff
@@ -1,12 +1,8 @@
-# ESP-IDF Partition Table
-# This gives us some additional space for code. 
-# It should also fix the OTA regression.  
-# Name,   Type, SubType,     Offset,      Size, Flags
-
-# Note that our NVS code assumes name 'storage' for the NVS partition
+# Standard 4M partition layout with OTA (Over the air updates)
 
+# Name,   Type, SubType,     Offset,      Size, Flags
 nvs,        data,   nvs,     0x00b000,    0x002000,
 otadata,    data,   ota,     0x00d000,    0x002000,
-app0,       app,    ota_0,   0x010000,    0x190000,
-app1,       app,    ota_1,   0x1A0000,    0x190000,
-storage,    data,   spiffs,  0x330000,    0x0D0000 
\ No newline at end of file
+app0,       app,    ota_0,   0x010000,    0x1A0000,
+app1,       app,    ota_1,   0x1B0000,    0x1A0000,
+storage,    data,   spiffs,  0x350000,    0x0B0000

```

Comment:
You'll note I stole another x20000 bytes from storage to make room for code, but it seems to still fit everything in!

Reply:
We can rename the file to include 4M, but I think the fact of the matter is that we (still) treat 4M as the default. The 8M table does include the 8M in the name. 
Anyway, I don't think I'd merge a PR that implements only this name change, but wouldn't block it if it was part of a bigger change to the way we set up our partitions.

Concerning not using 16MB partition tables, I think the honest reason for that is that there currently is no scenario where we really need more than 8MB. So, why complicate matters by introducing and maintaining another partition table? (That may sound like a bit of a lazy argument, but I think it's valid in its core. :) )

I've made a fiery argument for not destroying user's data partitions in a willy-nilly way in the past, and from a consumer product perspective I stand by that. At the same time and at this moment, the consumer product that Mesmerizer could potentially be is not sold that way; it's a kit that requires the buyer to somewhat know what they're doing to put it together. It's also not being sold by the thousands, to put it very mildly. 

Mesmerizer is also really the only project with effects that rely on configuration a user could care about, and far fewer effects allow themselves to be configured than I personally would have liked (hence #294, opened a year and a half ago yesterday). Off the top of my head, we're really only talking about the weather, stocks and _possibly_ subscribers effects.
I therefore think that _at the moment_ we can still make "breaking" changes to the partition layout _used by Mesmerizer_, but I reserve the right to change my mind further down the line if one or more of the parameters for my current perspective change. 
The partition layout for other projects can probably remain in development for (much) longer in any scenario.

And yes, I can see the benefit of a crash partition. If you're willing to pick this one up, then I'm happy to review it. In the end, the main criterion is that all projects still build and flash, and that all data we want to put in the data partitions fits in there. For most projects, that is "just" the web UI files and the JSON config and current effect files. The GIFs are only flashed for Mesmerizer, at the moment.

Reply:
Wait, hang on. _All_ flashed files are embedded into the firmware image, so end up on the application partition. The SPIFFS data partition really only contains the JSON config files (and the Improv log file, if that's enabled - which it normally isn't). 

I think that probably makes the challenge of making room for a crash partition a little more manageable...